quickwit-search/src/client.rs

```diff
@@ -121,6 +122,37 @@ impl SearchServiceClient {
         }
     }
 
+    /// Perform leaf export.
+    pub async fn leaf_export(
+        &mut self,
+        request: quickwit_proto::LeafExportRequest,
+    ) -> crate::Result<ReceiverStream<Result<LeafExportResult, tonic::Status>>> {
+        match &mut self.client_impl {
+            SearchServiceClientImpl::Grpc(grpc_client) => {
+                let tonic_request = Request::new(request);
+                let mut export_result_stream = grpc_client
+                    .leaf_export(tonic_request)
+                    .await
+                    .map_err(|tonic_error| parse_grpc_error(&tonic_error))?
+                    .into_inner();
+
+                // TODO: returning stream instead of a channel may be better.
+                // But this seems to be difficult. Try it at your own expense.
+                let (export_sender, export_receiver) = tokio::sync::mpsc::channel(10);
+                while let Some(result) = export_result_stream
+                    .message()
+                    .await
+                    .map_err(|status| parse_grpc_error(&status))?
```

Comment:
That looks good but I am surprsied of the type here.

Could we have ended with 
```
crate::Result<ReceiverStream<Result<LeafExportResult, tonic::Status>>>
```

Looking at the impl I get the impression that it is what it was doing, so I am confused.

Reply:
Yes so the stream returns result